subworkflows/local/assembly_qc.nf

```diff
@@ -64,17 +65,47 @@ workflow ASSEMBLY_QC {
         ch_versions      = ch_versions.mix(QUAST.out.versions)
     }
 
-    //
-    // Contiguate assembly with ABACAS
-    //
+//
+// Contiguate assembly with ABACAS
+//
     ch_abacas_results = Channel.empty()
     if (!params.skip_abacas) {
-        ABACAS (
+        fasta
+            .branch { fasta_file ->
+                def is_multi = false
+                def count = 0
+                file(fasta_file).withReader { reader ->
+                    String line
+                    while ((line = reader.readLine()) != null) {
+                        if (line.startsWith('>')) {
+                            count++
+                            if (count > 1) {
+                                is_multi = true
+                                break
+                            }
+                        }
+                    }
+                }
+                multi: is_multi
+                single: true
+            }
+            .set { fasta_type }
```

Comment:
Nextflow has an embedded function for this: https://www.nextflow.io/docs/latest/reference/operator.html#countfasta 
```suggestion
        fasta
            .branch { fasta ->
                multi: fasta.countFasta() > 1
                single: true
            }
            .set { fasta_type }
```